File: app/src/main/java/com/example/wgutscheduler/DAO/AssessmentDAO.java
package com.example.wgutscheduler.DAO;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.wgutscheduler.Entity.Assessment;

import java.util.List;

@Dao
public interface AssessmentDAO {
    @Query("SELECT * FROM assessment WHERE course_id_fk = :courseID ORDER BY assessment_id")
    List<Assessment> getAssessmentList(int courseID);

    @Query("Select * from assessment WHERE course_id_fk = :courseID and assessment_id = :assessmentID")
    Assessment getAssessment(int courseID, int assessmentID);

    @Query("SELECT * FROM assessment")
    List<Assessment> getAllAssessments();

    @Query("SELECT * FROM assessment WHERE course_id_fk = :courseID ORDER BY assessment_id DESC LIMIT 1")
    Assessment getCurrentAssessment(int courseID);

    @Insert
    void insertAssessment(Assessment assessment);

    @Insert
    void insertAllAssessments(Assessment... assessment);

    @Update
    void updateAssessment(Assessment assessment);

    @Delete
    void deleteAssessment(Assessment assessment);


}

File: app/src/main/java/com/example/wgutscheduler/DAO/AssessmentDAO.kt
package com.example.wgutscheduler.DAO

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.wgutscheduler.Entity.Assessment
import com.example.wgutscheduler.Entity.Term

@Dao
interface AssessmentDAO {
    @Query("SELECT * FROM assessment WHERE course_id_fk = :courseID ORDER BY assessment_id")
    fun getAssessmentList(courseID: Int): List<Assessment?>?

    @Query("Select * from assessment WHERE course_id_fk = :courseID and assessment_id = :assessmentID")
    fun getAssessment(courseID: Int, assessmentID: Int): Assessment?

    @get:Query("SELECT * FROM assessment")
    val allAssessments: List<Assessment?>?

    @Query("SELECT * FROM assessment WHERE course_id_fk = :courseID ORDER BY assessment_id DESC LIMIT 1")
    fun getCurrentAssessment(courseID: Int): Assessment?

    @Query("SELECT * FROM assessment WHERE assessment_name LIKE :assessmentName")
    fun searchAssessments(assessmentName: String): List<Assessment>

    @Query("SELECT term_id_fk FROM course INNER JOIN assessment ON course.course_id = assessment.course_id_fk WHERE course.course_id = :courseID")
    fun getTermId(courseID: Int): Int

    @Insert
    fun insertAssessment(assessment: Assessment?)

    @Insert
    fun insertAllAssessments(vararg assessment: Assessment?)

    @Update
    fun updateAssessment(assessment: Assessment?)

    @Delete
    fun deleteAssessment(assessment: Assessment?)
}